crates/router/src/connector/tsys/transformers.rs

```diff
@@ -215,7 +217,12 @@ fn get_payments_sync_response(
         mandate_reference: None,
         connector_metadata: None,
         network_txn_id: None,
-        connector_response_reference_id: None,
+        connector_response_reference_id: Some(
+            connector_response
+                .transaction_details
+                .transaction_id
+                .clone(),
+        ),
```

Comment:
```suggestion
        connector_response_reference_id: Some(
            connector_response
                .transaction_details
                .transaction_id,
        ),
```

Reply:
Hello @prasunna09 , Changing this piece of code throws an error with `cargo clippy --all-features`
![error](https://github.com/juspay/hyperswitch/assets/47860497/919d0ee1-0f81-4429-b57c-c1eeda988c17)

Reply:
We are still using the clone property here so that we don't stumble upon this error.

Reply:
My bad! Will approve this!